contracts/redpacket_erc721.sol

```diff
@@ -185,18 +210,12 @@ contract HappyRedPacket_ERC721 is Initializable {
 
     function refund(bytes32 id) external {
         RedPacket storage rp = redpacket_by_id[id];
-        Packed memory packed = rp.packed;
-        require(uint160(msg.sender) == unbox(packed.packed1, 0, 160), "Creator Only");
-        require(unbox(packed.packed2, 194, 32) <= block.timestamp, "Not expired yet");
-
-        uint256 remaining_tokens = unbox(packed.packed1, 160, 96);
+        require(msg.sender == rp.creator, "Creator Only");
+        require(rp.end_time <= block.timestamp, "Not expired yet");
+        uint256 remaining_tokens = rp.remaining_tokens;
```

Comment:
I think it might be better to unify the data type of `remaining_tokens`[less confusion]
Since the type of `rp.remaining_tokens` is `uint16`, maybe we should let the type of `remaining_tokens` in RefundSuccess be `uint16` as well.